include/effects/matrix/PatternWeather.h

```diff
@@ -222,50 +226,74 @@ class PatternWeather : public LEDStripEffect
     {
         HTTPClient http;
         String url = "http://api.openweathermap.org/data/2.5/forecast"
-            "?lat=" + strLatitude + "&lon=" + strLongitude + "&appid=" + urlEncode(g_ptrSystem->DeviceConfig().GetOpenWeatherAPIKey());
+            "?lat=" + strLatitude + "&lon=" + strLongitude + "&cnt=16&appid=" + urlEncode(g_ptrSystem->DeviceConfig().GetOpenWeatherAPIKey());
         http.begin(url);
         int httpResponseCode = http.GET();
 
         if (httpResponseCode > 0)
         {
-            AllocatedJsonDocument doc(4096);
+            // Needs to be this large to process all the returned JSON
+            AllocatedJsonDocument doc(10240);
             deserializeJson(doc, http.getString());
             JsonArray list = doc["list"];
 
             // Get tomorrow's date
-            time_t tomorrow = time(nullptr) + 86400;
-            tm* tomorrowTime = localtime(&tomorrow);
+            auto tomorrow = system_clock::now() + hours{24};
+            auto tomorrow_timet = system_clock::to_time_t(tomorrow);
+            auto tomorrowTime = localtime(&tomorrow_timet);
             char dateStr[11];
             strftime(dateStr, sizeof(dateStr), "%Y-%m-%d", tomorrowTime);
 
+            float localMin = 999.0;

```

Comment:
I'd stick to camelCase here. Concerning the brokenclouds_start etc, that's related to the GIF images that are stored in progmem. We use snake casing there because that's how the image start and end identifiers end up in the image as well, Because PlatformIO Says So.